### DISCUSSION TITLE
Creating 2d mesh

### DISCUSSION THREAD

**Discussion by JamesKeeF** [pmp-library/pmp-library#152] [Q&A]
Hey,

I could not see a clear example or documentation how this library works.

If I want to create a mesh for a rectangle with a hole (shaped like a circle) in the middle, what would I need to write in the code exactly?

As starting information I have the edge points of the rectangle, the center of the hole and the diameter. From what I've read, I understand that I need to create vertexes but what comes after that? 

Also if this is not possible with this library that is also good to know.

**Comment by dsieger** [comment]
Hi,

Indeed, this is not something you'd do with our library. We are more focused on processing existing meshes. You are looking for mesh generation. I'd recommend to checkout the following libraries:

Jonathan Shewchuck's [Triangle](https://www.cs.cmu.edu/~quake/triangle.html)
CGAL's [2D Triangulations](https://doc.cgal.org/latest/Triangulation_2/index.html), in particular constrained Delaunay triangulations

Hope this helps!
